release-notes/2.1/api-diff/2.0-vs-2.1_System.Collections.Specialized.md

```diff
@@ -0,0 +1,11 @@
+# System.Collections.Specialized
+
+``` diff
+ namespace System.Collections.Specialized {
+     public struct BitVector32 {
+-        public struct Section
++        public readonly struct Section
```

Comment:
@VSadov @stephentoub @jkotas can you guys refresh my memory is adding readonly to an existing struct a breaking change?

Reply:
No, as long as it's truthful and it actually is read-only.

Reply:
What defines it to be readonly? Do all the fields need to marked readonly?

Reply:
Yes, it is a nonbreaking if that is the only change.

Reply:
Adding `readonly` modifier will force that all instance fields/autoproperties/events are readonly.

If you have to change any of that, be careful!

Reply:
`readonly struct S1`    basically tells to the outer world "hey, I do not have any mutable state".

In a context of reference assemblies, - if a struct says it is readonly in a ref assembly, but an implementation adds a mutable instance field (even private one), it would be bad.

Reply:
Thanks. I just wanted to be sure that if someone was using it in some mutable way in the past and we now added readonly to that the compiler wouldn't start complaining.